linux_docker_resources/Dockerfile-RHEL

```
@@ -158,6 +158,7 @@ RUN dnf install \
     python3-rosdistro \
     python3-setuptools \
     python3-sip-devel \
+    python3-types-pyyaml \
```

Since this Dockerfile is used for both RHEL-8 (Humble) and RHEL-9 (Iron, Jazzy, and Rolling), we need to think about both.  In this case, it looks like `python3-types-pyyaml` is not available on RHEL-8, so this should be:

```suggestion
    $(if test ${EL_RELEASE/.*/} = 8; then echo python3-types-pyyaml; fi) \
```

---

Further (and this is annoying), we are going to have to add `types_pyyaml` to the `pip` installation below for RHEl-8, on lines https://github.com/ros2/ci/pull/781/files#diff-1903b2db128547ede4fad6a882d82becbbd9c06c57719d5543315b8006f031d5L198-L202 .  This is due to the fact that we can't use the `pip` install mechanism inside of `__main__.py` for RHEL-8 (pip has permissions problems that way).

---

Do we really need it on RHEL 8? This change isn't going to be backported to Humble, right?

---

> Do we really need it on RHEL 8? This change isn't going to be backported to Humble, right?

Oh yeah, actually, that's a great point.  So never mind that second comment, it is enough to just make sure we don't attempt to install it on RHEL-8 (so just accept my suggestion, and that should be good enough).

---

@clalancette I think your conditional is inverted.

---

Sigh, yes, sorry.